lib/Doctrine/Common/Cache/FileCache.php

```diff
@@ -126,9 +113,9 @@ protected function getFilename($id)
     {
         return $this->directory
             . DIRECTORY_SEPARATOR
-            . implode(str_split(hash('sha256', $id), 2), DIRECTORY_SEPARATOR)
+            . substr(hash('sha256', $id), 0, 2)
             . DIRECTORY_SEPARATOR
-            . preg_replace($this->disallowedCharacterPatterns, $this->replacementCharacters, $id)
+            . rawurlencode($id)
```

Comment:
> This is probably also one of the reasons the hash above was introduced.

Agreed, although the hash also guarantees some better distribution of the files across directories (otherwise, this cache has a hard limit imposed by the number of files per directory supported by the FS)